src/core/viz/shader-compiler.js

```diff
@@ -1,4 +1,19 @@
+class IDGenerator {
```

Comment:
this should be moved to its own file

Reply:
Disagreed, this is just a 34 line long file, and it is logically coupled here. The project doesn't follow the 1 class - 1 file rule too.